yield_plugin/src/yield_plugin.cpp

```diff
@@ -37,72 +37,344 @@ using oss = std::ostringstream;
+        cav_msgs::Trajectory incoming_trajectory = incoming_request.trajectory;
+        std::string req_strategy_params = incoming_request.strategy_params;
+        
+        
+
+        // Parse strategy parameters
+        using boost::property_tree::ptree;
+        ptree pt;
+        std::istringstream strstream(req_strategy_params);
+        boost::property_tree::json_parser::read_json(strstream, pt);
+        double req_traj_speed = pt.get<double>("speed");
+        int start_lanelet_id = pt.get<int>("start_lanelet");
+        int end_lanelet_id = pt.get<int>("end_lanelet");
+
+
+        std::vector<lanelet::BasicPoint2d> req_traj_plan = {};
+        try
+        {
+          geometry_msgs::TransformStamped tf = tf2_buffer_.lookupTransform("map", "earth", ros::Time(0));
+          req_traj_plan = convert_eceftrajectory_to_mappoints(incoming_trajectory, tf);
+        }
+        catch (const tf2::TransformException &ex)
+        {
+            ROS_WARN("%s", ex.what());
+        }
+
+        double req_expiration_sec = incoming_request.expiration/1000;
+        double current_time = ros::Time::now().toSec();
+
+        cav_msgs::MobilityResponse outgoing_response = compose_mobility_response(req_sender_id, req_plan_id);
+        // ensure there is enough time for the yield
+        if (req_expiration_sec - current_time >= config_.tpmin)
+        {
+          received_cooperative_request_ = true;
+          double req_plan_time = req_expiration_sec - current_time;
+          set_incoming_request_info(req_traj_plan, req_traj_speed, req_plan_time);
+          lc_status_msg.status = cav_msgs::LaneChangeStatus::REQUEST_ACCEPTED;
+          lc_status_msg.description = "Cooperative Lane Change Request Accepted by Yield Plugin";
+        
+        }
+        else
+        {
+          ROS_DEBUG_STREAM("Igonore expired Mobility Request.");
+          outgoing_response.is_accepted = false;
+          lc_status_msg.status = cav_msgs::LaneChangeStatus::REQUEST_REJECTED;
+          lc_status_msg.description = "Cooperative Lane Change Request Rejected by Yield Plugin";
+        }
+        mobility_response_publisher_(outgoing_response);
+        lc_status_msg.status = cav_msgs::LaneChangeStatus::RESPONSE_SENT;
+        lc_status_msg.description = "Cooperative Lane Change Response Sent";
+      }
+    }
+    if (lanechange_status_pub_.isLatched())
+    {
+      lanechange_status_pub_.publish(lc_status_msg);
+    }
+    else
+    {
+      ROS_WARN("Lane Change Status Topic is not latched.")
+    }
+    
+  }
+
+  void YieldPlugin::set_incoming_request_info(std::vector <lanelet::BasicPoint2d> req_trajectory, double req_speed, double req_planning_time)
+  {
+    req_trajectory_points_ = req_trajectory;
+    req_target_speed_ = req_speed;
+    req_target_plan_time_ = req_planning_time;
+  }
+
+
+  void YieldPlugin::bsm_cb(const cav_msgs::BSMConstPtr& msg)
+  {
+    cav_msgs::BSMCoreData bsm_core_ = msg->core_data;
+    host_bsm_id_ = bsmIDtoString(bsm_core_);
```

Comment:
Not sure I understand the naming, is this actually the host vehicle's BSM ID? Or would this be the ID of the vehicle sending us the request? if the latter, we should probably rename this for clarity.